src/entities.py

```diff
@@ -95,193 +95,241 @@ def __hash__(self) -> int:
-        "Y": "Ei",
-        "Z": "Wt",
-    }
-
-    def __init__(self, location: str):
-        self.location = location
-        self.ingredient_set = set()
-
-    def _values_lookup(self, values: Sequence[str], lookup: Optional[Dict[str, str]]) -> None:
-        """
-        Normalizes ingredients to the self.ingredient_lookup codes.
-
-        Args:
-            values: A sequence of ingredients codes.
-            lookup: If needed, a mapping from a canteen specific ingredients codes to the self.ingredient_lookup codes.
-        """
-        for value in values:
-            # ignore empty values
-            if not value or value.isspace():
-                continue
-            if (not lookup and value not in self.ingredient_lookup) or (lookup and value not in lookup):
-
-                # sometimes the ‘,’ is missing between the ingredients (especially with IPP) and we try to split again
-                # with capital letters.
-                split_values: List[Any] = re.findall(r"[a-züöäA-ZÜÖÄ][^A-ZÜÖÄ]*", value)
-                if split_values:
-                    self._values_lookup(split_values, lookup)
-                    continue
-                else:
-                    print("Unknown ingredient for " + self.location + " found: " + str(value))
-                    continue
-
-            if lookup:
-                self.ingredient_set.add(lookup[value])
-            else:
-                self.ingredient_set.add(value)
-
-    def parse_ingredients(self, values: str) -> None:
-        """
-        Parse and creates a normalized list of ingredients.
-
-        Args:
-            values: String with comma separated ingredients codes.
-        """
-        values = values.strip()
-        split_values: List[str] = values.split(",")
-        # check for special parser/ingredient translation required
-        if self.location == "fmi-bistro":
-            self._values_lookup(split_values, self.fmi_ingredient_lookup)
-        elif self.location == "mediziner-mensa":
-            self._values_lookup(split_values, self.mediziner_ingredient_lookup)
-        # default to the "Studentenwerk" ingredients
-        # "ipp-bistro" also uses the "Studentenwerk" ingredients since all
-        # dishes contain the same ingredients
-        else:
-            self._values_lookup(split_values, None)
+class Site:
+    def __init__(self, address: str, latitude: float, longitude: float):
+        self.address = address
+        self.latitude = latitude
+        self.longitude = longitude
 
-    def __hash__(self) -> int:
-        return hash(frozenset(self.ingredient_set))
+    def to_json_obj(self):
+        return {
+            "address": self.address,
+            "latitude": self.latitude,
+            "longitude": self.longitude,
+        }
+
+
+class Location(Enum):
+    # Some of the locations do not use the general Studentenwerk system and therefore do not have a url_id.
+    def __init__(self, long_name: str, site: Site, url_id: int):
```

Comment:
We should add an canteen_id, to have the same functionality as before.

Reply:
Which functionality do you mean?

Reply:
As far as I'm concerned, the `canteen_id` is used, to identify the folder containing the dishes for the specific canteen.
So for example the "Mensa Garching" has the canteen_id `mensa-garching`, and in the gh-pages (so the API) you can find it in the path with `mensa-garching`: https://github.com/TUM-Dev/eat-api/tree/gh-pages/mensa-garching

Further, this is also required for the website, or other programs to function properly.
This has in mind, that the `canteens.json` should be removed and instead generated through the enum.

Reply:
Ok, I'll add it.